session.go

```diff
@@ -331,6 +332,21 @@ func (sess *session) handleRequests(reqs <-chan *gossh.Request) {
 					continue
 				}
 			}
+			if err := ptyReq.allocate(); err != nil {
+				req.Reply(false, nil)
+				continue
+			}
+			if ptyReq.Pty != nil {
+				log.Printf("pty allocated: %s", ptyReq.Pty.Name())
+				sess.env = append(sess.env,
+					fmt.Sprintf("TERM=%s", ptyReq.Term),
+					fmt.Sprintf("SSH_TTY=%s", ptyReq.Pty.Name()),
+				)
```

Comment:
I think its safer to set those vars first, so we make sure the user didn't somehow set them?

```suggestion
				sess.env = append(
				  []string{
				    fmt.Sprintf("TERM=%s", ptyReq.Term),
					  fmt.Sprintf("SSH_TTY=%s", ptyReq.Pty.Name()),
				  },
				  sess.env...,
				)
```